hosting.go

```diff
@@ -143,24 +143,34 @@ type HostingEmail struct {
 	Link HostingLink `json:"_links"`
 }
 
+type HostingListOptions struct {
```

Comment:
Instead of having those limit and offset fields everywhere, I would suggest to have a single pagination struct for all API
```
type PaginationOptions struct {
    Limit  *int `param:"limit"`
    Offset *int `param:"offset"`
}
```
which is then used directly (to get rid altogether of HostingListEmailAliasesOptions, etc... for example)
or embedded
```
type HostingListOptions struct {
    PaginationOptions
    Type *string `param:"type"`
}
```
I'm ok if it's done in this PR or in a separate one.

Reply:
Done